filters/openpolicyagent/evaluation.go

```diff
@@ -0,0 +1,76 @@
+package openpolicyagent
+
+import (
+	"context"
+	"time"
+
+	ext_authz_v3 "github.com/envoyproxy/go-control-plane/envoy/service/auth/v3"
+	"github.com/open-policy-agent/opa-envoy-plugin/envoyauth"
+	"github.com/open-policy-agent/opa-envoy-plugin/opa/decisionlog"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/server"
+	"github.com/open-policy-agent/opa/tracing"
+	"github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+)
+
+func (opa *OpenPolicyAgentInstance) Eval(ctx context.Context, req *ext_authz_v3.CheckRequest) (*envoyauth.EvalResult, error) {
+	var err error
+
+	result, stopeval, err := envoyauth.NewEvalResult()
+	span := opentracing.SpanFromContext(ctx)
+	if span != nil {
+		span.SetTag("decision_id", result.DecisionID)
+	}
+
+	if err != nil {
+		opa.Logger().WithFields(map[string]interface{}{"err": err}).Error("Unable to generate decision ID.")
+		return nil, err
+	}
+
+	var input map[string]interface{}
+	defer func() {
+		stopeval()
+		err := opa.logDecision(ctx, input, result, err)
+		if err != nil {
+			opa.Logger().WithFields(map[string]interface{}{"err": err}).Error("Unable to log decision to control plane.")
+		}
+	}()
+
+	if ctx.Err() != nil {
+		err = errors.Wrap(ctx.Err(), "check request timed out before query execution")
+		return nil, err
+	}
+
+	logger := opa.manager.Logger().WithFields(map[string]interface{}{"decision-id": result.DecisionID})
+	input, err = envoyauth.RequestToInput(req, logger, nil, true)
+	if err != nil {
+		return nil, err
+	}
+
+	inputValue, err := ast.InterfaceToValue(input)
+	if err != nil {
+		return nil, err
+	}
+
+	err = envoyauth.Eval(ctx, opa, inputValue, result, rego.DistributedTracingOpts(tracing.Options{opa}))
```

Comment:
I see where it comes from but it reads a bit weird to do `envoyauth` and not `opaauth`. I guess we can rename the import, not sure if it makes sense to everyone to do my proposed change, but personally I would wonder why we do something envoy as it is also an http proxy and we do not want to run 2 proxies.

Reply:
We could also use `opaenvoyplugin.Eval`. `opaauth` might be confusing as well as this is not from the core Go library but then you still can just navigate to the actual target.